lib/tasks/book2.rake

```diff
@@ -98,10 +98,10 @@ task :genbook2 => :environment do
           html.gsub!('<h1', '<h2')
           html.gsub!(/\/h1>/, '/h2>')
 
-          if xlink = html.scan(/\.html\#(.*?)\"/)
+          if xlink = html.scan(/href=\"1-.*?\.html\#(.*?)\"/)
```

Comment:
This tightens up the regex, so I wondered if we would miss any other cases that did not point into the `1-` files. As it turns out there is only one such match, and it is a false positive that your change fixes! In A1.5 ("Git in Zsh"), the link to:

```
http://zsh.sourceforge.net/Doc/Release/User-Contributions.html#Version-Control-Information
```

is currently mangled into:

```
http://zsh.sourceforge.net/Doc/Release/User-Contributions/Version-Control-Information
```

which naturally gives a 404. With your patch, it's preserved correctly.